CONTRIBUTING.md

```diff
@@ -1,16 +1,281 @@
+development workflow on GitHub.
+
+[try-github-io]: https://try.github.io/
+[git-scm-getting-started]: https://git-scm.com/book/en/v2/Getting-Started-Git-Basics
+
+#### Setup your local RIOT repository
+
+Before you start modifying code, you need to fork the RIOT upstream repository
+from the [RIOT main GitHub page][riot-github].
+
+If it's your first time with git, configure your name and emails:
+
+    $ git config --global user.name = "<your name here>"
+    $ git config --global user.email = "<your email address here>"
+
+Then clone locally your fork of RIOT (replace `account name` with your actual
+login on GitHub):
+
+    $ git clone git@github.com:<account name>/RIOT.git
+
+You can keep any branch of your local repository up-to-date with the upstream
+master branch with the following commands:
+
+    $ git checkout <branch name>
+    $ git pull --rebase https://github.com/RIOT-OS/RIOT.git
+
+Use it before opening a PR. This will at least ensure the PR is mergeable but
+also that it is up-to-date with the upstream repository.
+
+[riot-github]: https://github.com/RIOT-OS/RIOT
+
+#### Work on branches
+
+Avoid opening PR from the `master` branch of your fork to the master branch of
+the RIOT upstream repository: update your master branch and start a new branch
+from it.
+
+    $ git checkout master
+    $ git pull --rebase https://github.com/RIOT-OS/RIOT.git
+    $ git checkout -b <new branch>
+    # Do your changes, commit, update with latest upstream master
+    $ git push
+
+#### Add fixup commits during review
+
+To keep the history of changes easier to track for reviewers, it is recommended
+to push your review request updates in fixup commits.
+
+Let's say your PR contains 3 commits with comments: `prefix1: change 1`,
+`prefix2: change 2` and `prefix3: change 3`.
+
+Instead of committing changes in `prefix2` in a 4th commit `prefix2: change 4`,
+you can use the `--fixup` option:
+
+    $ git add /path/of/prefix2
+    $ git commit --fixup <prefix2 commit hash>
+
+#### Squash commits after review
+
+Squashing a commit is done using the rebase subcommand of git in interactive
+mode:
+
+    $ git rebase master -i
+
+You can find information on rebasing in
+[GitHub rebase documentation][about-git-rebase].
+
+[about-git-rebase]: https://help.github.com/articles/about-git-rebase/
+
+If you used [fixup commits](#add-fixup-commits-during-review) during the review
+phase, squashing commits can be performed in a single command:
+
+    $ git rebase -i --autosquash
+
+**Watch out: Don't squash your commit until a maintainer asks you to do it.**
+
+Otherwise the history of review changes is lost and for large PRs, it
+makes it difficult for the reviewer to follow them. It might also happen that
+you introduce regression and won't be able to recover them from previous
+commits.
+
+Once squashing is done, you will have to force push your branch to update the
+PR:
+
+    $ git push -f
```

Comment:
I'd rather recommend [`git push --force-with-lease`](https://git-scm.com/docs/git-push#git-push---no-force-with-lease) to newcomers ;-) (there are some nice articles on the web that explain better than me why this is better).